src/hooks/use-document-sync-to-firebase.ts

```diff
@@ -169,13 +170,28 @@ export function useDocumentSyncToFirebase(
   const mutation = useMutation((snapshot: DocumentContentSnapshotType) => {
     const tileMap = snapshot.tileMap || {};
 
-    const tileTypes: string[] = [];
+    const tools: string[] = [];
 
     Object.keys(tileMap).forEach((tileKey) => {
       const tileInfo = tileMap[tileKey] as ITileMapEntry;
       const tileType = tileInfo.content.type;
-      if (!tileTypes.includes(tileType)) {
-        tileTypes.push(tileType);
+      if (!tools.includes(tileType)) {
+        tools.push(tileType);
+      }
+    });
+
+    // The annotations property does exist on the snapshot but MobX doesn't recognize it
+    // as a property because of the way we are constructing the DocumentContentModel
+    // on top of multiple other models. This typing is a workaround so TS doesn't complain.
+    const annotations =
+      (snapshot as {annotations: Record<string, IArrowAnnotation>}).annotations || {};
+
+    Object.keys(annotations).forEach((annotationKey: string) => {
+      const annotation = annotations[annotationKey];
+      // for now we only want Sparrow annotations
+      // we might want to change this if we want to count other types in the future
+      if (!tools.includes(annotation.type) && annotation.type === "arrowAnnotation") {
```

Comment:
This looks like an error. It is checking if tools includes annotation.type which will likely be "arrowAnnotation" and then it is added "Sparrow" to the tools. So the next time through the loop if there is another annotation it will add "Sparrow" again.